### PR TITLE
Remove slf4j-impl (#6364)

### PR DESCRIPTION
**Pull request by skhatri**
## Description
Add a description of your PR here.
Fix logging dependencies in pinot-spi #6364
## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)
## Release Notes
If you have tagged this as either backward-incompat or release-notes,
you MUST add text here that you would like to see appear in release notes of the
next release.

If you have a series of commits adding or enabling a feature, then
add this section only in final commit that marks the feature completed.
Refer to earlier release notes to see examples of text

## Documentation
If you have introduced a new feature or configuration, please add it to the documentation as well.
See https://docs.pinot.apache.org/developers/developers-and-contributors/update-document

### CODE REVIEW

**Comment by skhatri** [comment]
dependency tree(mvn -pl pinot-spi dependency:tree) with slf4j-api 
[INFO] org.apache.pinot:pinot-spi:jar:0.7.0-SNAPSHOT
[INFO] +- commons-configuration:commons-configuration:jar:1.10:compile
[INFO] +- commons-codec:commons-codec:jar:1.6:compile
[INFO] +- commons-io:commons-io:jar:2.4:compile
[INFO] +- commons-lang:commons-lang:jar:2.6:compile
[INFO] +- commons-logging:commons-logging:jar:1.2:compile
[INFO] +- org.apache.commons:commons-lang3:jar:3.5:compile
[INFO] +- commons-collections:commons-collections:jar:3.2.1:compile
[INFO] +- com.google.guava:guava:jar:20.0:compile
[INFO] +- com.google.code.findbugs:jsr305:jar:3.0.0:compile
[INFO] +- org.slf4j:slf4j-api:jar:1.7.25:compile
[INFO] +- com.lmax:disruptor:jar:3.3.4:compile
[INFO] +- org.apache.logging.log4j:log4j-1.2-api:jar:2.11.2:compile
[INFO] |  +- org.apache.logging.log4j:log4j-api:jar:2.11.2:compile
[INFO] |  \- org.apache.logging.log4j:log4j-core:jar:2.11.2:compile
[INFO] +- joda-time:joda-time:jar:2.10.5:compile
[INFO] +- com.fasterxml.jackson.core:jackson-annotations:jar:2.10.0:compile
[INFO] +- org.yaml:snakeyaml:jar:1.16:compile
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.10.0:compile
[INFO] |  \- com.fasterxml.jackson.core:jackson-core:jar:2.10.0:compile
[INFO] +- org.apache.avro:avro:jar:1.7.6:compile
[INFO] |  +- org.codehaus.jackson:jackson-core-asl:jar:1.9.13:compile
[INFO] |  +- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.13:compile
[INFO] |  +- com.thoughtworks.paranamer:paranamer:jar:2.8:compile
[INFO] |  +- org.xerial.snappy:snappy-java:jar:1.1.1.7:compile
[INFO] |  \- org.apache.commons:commons-compress:jar:1.20:compile
[INFO] +- org.codehaus.groovy:groovy-all:jar:2.4.21:compile
[INFO] +- org.testng:testng:jar:6.11:test
[INFO] |  \- com.beust:jcommander:jar:1.64:test
[INFO] \- org.reflections:reflections:jar:0.9.11:compile
[INFO]    \- org.javassist:javassist:jar:3.19.0-GA:compile

File: pinot-plugins/pinot-file-system/pinot-s3/pom.xml [hunk 1]
```diff
@@ -151,6 +152,12 @@
       <artifactId>netty-common</artifactId>
       <version>${netty.version}</version>
     </dependency>
+    <dependency>
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-file-system/pinot-s3/pom.xml:155]: Why do we need this dependency?

Comment (jackjlli) [reply pinot-plugins/pinot-file-system/pinot-s3/pom.xml:155]: +1 on that.

**Comment by KKcorps** [comment]
Closing this PR.

**Comment by kkrugler** [comment]
Hi @KKcorps - why was this closed? Removing direct dependencies on `slf4j-impl` in lower-level jars is a good thing, usually. Or did this get covered by a different PR?

**Comment by KKcorps** [comment]
@kkrugler I closed it for inactivity. I am not sure if this change is needed. Opening it again  for review.

**Comment by kkrugler** [comment]
@KKcorps - A jar that has a `main()` method that's used from the command line needs a logging implementation. Everything else should only be using the `slf4j-api`. Otherwise (e.g. with the Hadoop segment building workflow, which uses various Pinot jars) there can be logging implementation collisions. I made a run at fixing this up, but it got very complicated due to other 3rd party libraries used by Pinot pulling in their own logging implementations, which necessitates using various xxx-to-slf4j bridges. Right now it's kind of a mess....

**Comment by KKcorps** [comment]
@kkrugler - Makes sense. So what this PR does for s3 plugin has to be repeated for every other plugin as well, right?

**Comment by kkrugler** [comment]
@KKcorps - yes, exactly. Plus other (lower-level) Pinot jars that include logging implementations.

**Pull request closed without merging**